File: 02-PatronesArrayString/01-VerifyingAlienDictionary.js
word = ["habito", "hacer", "lectura", "sonreir"];
orden_alfabeto = "hlabcdefgijkmnopqrstuvwxyz";

function verifying_alien(words, orden_alfabeto) {
  const count_array = words.length;
  const array_alphabeth = Array.from(orden_alfabeto);

  for (let i = 1; i < count_array; i++) {
    let previusWord = [...words[i - 1]];
    let nextWord = [...words[i]];
    let len_previusWord = previusWord.length;
    let len_nextWord = nextWord.length;
    let less_letter = 0;

    if (len_previusWord < len_nextWord) {
      less_letter = len_previusWord;
    } else {
      less_letter = len_nextWord;
    }

    for (let y = 0; y < less_letter; y++) {
      if (previusWord[y] == nextWord[y]) {
        if (len_nextWord < len_previusWord && y + 1 == less_letter) {
          return false;
        }
      } else if (
        array_alphabeth.findIndex((letter) => letter === previusWord[y]) <
        array_alphabeth.findIndex((letter) => letter === nextWord[y])
      ) {
        break;
      } else {
        return false;
      }
    }
  }
  return true;
}
console.log(verifying_alien(word, orden_alfabeto));

File: 02-PatronesArrayString/01Patrones Lineales/01-VerifyingAlienDictionary.py
word = ["habito", "hacer", "lectura", "sonreir"]
orden_alfabeto = "hlabcdefgijkmnopqrstuvwxyz"

def verifying_alien(array,orden_alfabeto):
    count_array= len(array)
    array_alphabeth= list(orden_alfabeto)

    for i in range (1,count_array):
        previusWord= list(word[i-1])    
        nextWord= list(word[i])    
        len_previusWord= len(previusWord)
        len_nextWord= len(nextWord)
        less_letter=0

        if (len_previusWord<len_nextWord):
            less_letter=len_previusWord
        else:
            less_letter=len_nextWord

        for y in range (less_letter):
            if (previusWord[y]==nextWord[y]):
                if (len_nextWord<len_previusWord and (y+1)==less_letter):
                    return False
            elif (array_alphabeth.index(previusWord[y])<array_alphabeth.index(nextWord[y])):
                break
            else:
                return False
    return True

print(verifying_alien(word,orden_alfabeto))
